packages/flutter/lib/src/widgets/view.dart

```diff
@@ -106,13 +145,669 @@ class View extends StatelessWidget {
+  /// given `context`.
+  ///
+  /// Calling this method establishes a dependency and the provided `context`
+  /// and causes it to rebuild whenever the [ViewHooks] change.
+  ///
+  /// When no [ViewHooks] are available in the provided `context`, a default
+  /// ViewHooks instance with [renderViewManager] set to
+  /// [RendererBinding.instance] and [pipelineOwner] set to
+  /// [RendererBinding.rootPipelineOwner] is returned.
+  static ViewHooks of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<ViewHooksScope>()?.hooks
+        ?? ViewHooks(
+          renderViewManager: RendererBinding.instance,
+          pipelineOwner: RendererBinding.instance.rootPipelineOwner,
+        );
+  }
+
+  /// The [RenderViewManager] to which the [RawView] widget should add the
+  /// [RenderView] root of its render tree by calling
+  /// [RenderViewManager.addRenderView].
+  final RenderViewManager renderViewManager;
+
+  /// The parent [PipelineOwner] to which the [RawView] widget should add the
+  /// [PipelineOwner] managing its render tree by calling
+  /// [PipelineOwner.adoptChild].
+  final PipelineOwner pipelineOwner;
+
+  /// Create a clone of the current [ViewHooks] but with provided parameters
+  /// replaced.
+  ViewHooks copyWith({
+    RenderViewManager? renderViewManager,
+    PipelineOwner? pipelineOwner,
+  }) {
+    assert(renderViewManager != null || pipelineOwner != null);
+    return ViewHooks(
+      renderViewManager: renderViewManager ?? this.renderViewManager,
+      pipelineOwner: pipelineOwner ?? this.pipelineOwner,
+    );
+  }
+
+  @override
+  int get hashCode => Object.hash(renderViewManager, pipelineOwner);
+
+  @override
+  bool operator ==(Object other) {
+    return other is ViewHooks
+        && renderViewManager == other.renderViewManager
+        && pipelineOwner == other.pipelineOwner;
+  }
+
+  @override
+  String toString() {
+    return
+      '${objectRuntimeType(this, 'ViewHooks')}('
+        'pipelineOwner: $pipelineOwner, '
+        'renderViewManager: $renderViewManager'
+      ')';
+  }
+}
+
+class _MultiChildComponentWidget extends Widget {
+  const _MultiChildComponentWidget({
+    super.key,
+    List<Widget> views = const <Widget>[],
+    Widget? child,
+  }) : _views = views, _child = child;
+
+  // It is up to the subclasses to make the relevant properties public.
+  final List<Widget> _views;
+  final Widget? _child;
+
+  @override
+  Element createElement() => _MultiChildComponentElement(this);
+}
+
+/// A collection of sibling [View]s.
+///
+/// The widget can only be used in places were a [View] widget is legal. In
+/// practical terms, it can be used at the root of the widget tree outside of
+/// any [View] widget, as a child to a another [ViewCollection], or in the
+/// [ViewAnchor.view] slot of a [ViewAnchor] widget. It is not required to be a
+/// direct child of those widgets; other non-[RenderObjectWidget]s may appear
+/// in between the two.
```

Comment:
```suggestion
/// in between the two (such as an [InheritedWidget]).
```
(although I guess you do say it in the next paragraph, so maybe not needed)

Reply:
Added it anyways. I think it gives a little more context to what non-RenderObjectWidgets may be.